pallets/nft/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+			let sender = ensure_signed(origin)?;
+			let _class_info = orml_nft::Pallet::<T>::classes(token.0).ok_or(Error::<T>::ClassNotFound)?;
+			let token_info = orml_nft::Pallet::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenNotFound)?;
+			ensure!(sender == token_info.owner, Error::<T>::NoPermission);
+			ensure!(!token_info.data.locked, Error::<T>::TokenLocked);
+			orml_nft::Pallet::<T>::burn(&sender, token)?;
+			Self::deposit_event(Event::NFTTokenBurned(sender, token.0, token.1));
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::destroy_class())]
+		pub fn destroy_class(origin: OriginFor<T>, class_id: T::ClassId) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			let class_info = orml_nft::Pallet::<T>::classes(class_id).ok_or(Error::<T>::ClassNotFound)?;
+			ensure!(sender == class_info.owner, Error::<T>::NoPermission);
+			ensure!(
+				class_info.total_issuance == Zero::zero(),
+				Error::<T>::CannotDestroyClass
+			);
+			orml_nft::Pallet::<T>::destroy_class(&sender, class_id)?;
+			Self::deposit_event(Event::NFTTokenClassDestroyed(sender, class_id));
+			Ok(().into())
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		NFTTokenClassCreated(T::AccountId, T::ClassId),
+		NFTTokenMinted(T::AccountId, T::ClassId, u32),
+		NFTTokenMintedLockToggled(T::AccountId, T::ClassId, T::TokenId, bool),
+		NFTTokenTransferred(T::AccountId, T::AccountId, T::ClassId, T::TokenId),
+		NFTTokenBurned(T::AccountId, T::ClassId, T::TokenId),
+		NFTTokenClassDestroyed(T::AccountId, T::ClassId),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		ClassNotFound,
+		TokenNotFound,
+		NoPermission,
+		CannotDestroyClass,
+		TokenLocked,
+		InvalidQuantity,
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn is_owner(account: &T::AccountId, token: (T::ClassId, T::TokenId)) -> bool {
+		orml_nft::Pallet::<T>::is_owner(account, token)
+	}
+
+	pub fn is_locked(token: (T::ClassId, T::TokenId)) -> Result<bool, DispatchError> {
+		let token_info = orml_nft::Pallet::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenNotFound)?;
+		Ok(token_info.data.locked)
+	}
+
+	pub fn toggle_lock(account: &T::AccountId, token_id: (T::ClassId, T::TokenId)) -> DispatchResult {
+		let _class_info = orml_nft::Pallet::<T>::classes(token_id.0).ok_or(Error::<T>::ClassNotFound)?;
+		orml_nft::Tokens::<T>::mutate_exists(token_id.0, token_id.1, |token| -> DispatchResult {
+			if let Some(ref mut token) = token {
+				ensure!(*account == token.owner, Error::<T>::NoPermission);
+				token.data.locked ^= true; // Toggle
+						   // fix clone
+				Self::deposit_event(Event::NFTTokenMintedLockToggled(
+					account.clone(),
+					token_id.0,
+					token_id.1,
+					token.data.locked,
+				));
+			}
+			Ok(())
+		})?;
```

Comment:
I believe that you can just return from here directly. So the last Ok(()) won;t be necessary.

Reply:
thought so too, but doesn't work

Reply:
```suggestion
		})
```

and removing the Ok(()) at the of the function  - that should be fine.